### ISSUE TITLE
Recent branches randomly breaks?

### ISSUE THREAD

**Issue by joshaber** [desktop/desktop#2426]
This is something I've noticed off and on. _Sometimes_ I'll end up with no recent branches.

Unclear what's going on here.

**Comment by joshaber** [comment]
This is a problem at the command line level too. `git reflow` seems to be truncated.

**Comment by joshaber** [comment]
This seems to be related to renames. The reflog truncates at any rename entry.

```
86af651fa94ab0d1e64c716bc1c648cb6f0d695a 0000000000000000000000000000000000000000 joshaber <joshaber@gmail.com> 1502292974 -0400	Branch: renamed refs/heads/this-is-a-test to refs/heads/this-is-a-test-also
0000000000000000000000000000000000000000 86af651fa94ab0d1e64c716bc1c648cb6f0d695a joshaber <joshaber@gmail.com> 1502292974 -0400	Branch: renamed refs/heads/this-is-a-test to refs/heads/this-is-a-test-also
```

**Comment by joshaber** [comment]
This looks like it might be a bug in our version of git.

**Comment by joshaber** [comment]
Aha! Looks like it was fixed in 2.13.3:

https://github.com/git/git/blob/4384e3cde2ce8ecd194202e171ae16333d241326/Documentation/RelNotes/2.13.3.txt#L53-L56

**Comment by joshaber** [comment]
Created a PR to bump dugite: https://github.com/desktop/dugite/pull/109

**Comment by shiftkey** [comment]
@joshaber I pushed #2432 to test for any regressions, but if you want to add a regression test to close this out it'd be :gem:

**Comment by shiftkey** [comment]
Fixed by #2432
